src/plugins/obj/s3/engine_impl.cpp

```
@@ -80,17 +80,28 @@ class nixlObjBackendReqH : public nixlBackendReqH {
 
 class nixlObjMetadata : public nixlBackendMD {
 public:
+    // Object segment: maps a device id to an S3 object key.
     nixlObjMetadata(nixl_mem_t nixl_mem, uint64_t dev_id, std::string obj_key)
```

Cppcheck (2.20.0)

[performance] Function parameter 'obj_key' should be passed by const reference.

(passedByValue)